### PR TITLE
StateTimeline: Support hideFrom field config

### PR DESCRIPTION
**Pull request by ryantxu**
This updates the state timeline and status history panels so that hide from "viz" works as expected:

The legend is still a bit weird, but i think works as well as it can for now

### CODE REVIEW

File: .yarn/sdks/prettier/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "prettier",
-  "version": "2.8.1-sdk",
+  "version": "2.8.4-sdk",
```
Comment (nmarrs) [inline .yarn/sdks/prettier/package.json:3]: is this change needed?

**Review by nmarrs — APPROVED** [review]
Legend + tooltip seem to not work quite as expected - though we can push that off to another issue / PR if desired.

https://github.com/grafana/grafana/assets/22381771/7475087c-1ad9-4a37-a933-d6ae381f562c

**Comment by ryantxu** [comment]
> Legend + tooltip seem to not work quite as expected

ya -- the cases are odd there so no easy fix.  I vote we merge this, then improve.  

The "legend" option depends so much on what it is showing for it to make sense.  By default it is showing thresholds.  It will work as expected with explicit value mapping.  And it totally ignored for continuous 😵‍💫

File: .yarn/sdks/prettier/package.json (hunk 1 above)
Comment (ryantxu) [reply .yarn/sdks/prettier/package.json:3]: no... but also don't want to make an new PR for it!
